python/rmm/_lib/device_buffer.pyx

```diff
@@ -81,6 +84,10 @@ cdef class DeviceBuffer:
                 if stream.c_is_default():
                     stream.c_synchronize()
 
+        # Save a reference to the MR and stream used for allocation
+        self.mr = get_current_device_resource()
```

Comment:
Do we want to allow passing a MR to the constructor?

Reply:
For now, I'd say let's not do that. I'd much rather we support multiple MRs with a context manager. That is, I prefer the first approach below to the second:

```python
mr = Memory resource(...)
with using_memory_resource(mr)
    dbuf_1 = DeviceBuffer(...)
    dbuf_2 = DeviceBuffer(...)
```

v/s

```python
mr = MemoryResource(...)
dbuf_1 = DeviceBuffer(..., mr=mr)
dbuf_2 = DeviceBuffer(..., mr=mr)
```

There are arguments in favour of both though.

Reply:
The context manager approach also follows the "One -- and preferably only one -- obvious way to do it" Zen.

Reply:
For now, I have left out passing the MR in the constructor. If the situation has changed, @kkraus14 or @shwina please let me know.